### PR TITLE
Add namespace as target for @access decorator

### LINKED ISSUE #1299: Have `@access` decorator apply to namespaces
### Clear and concise description of the problem

We have `@access` for individual models and properties: we want to be able to apply this globally through a decorator that takes in a namespace / interface. More granularity down the line wins out

```md
| Decorator State            | Model with Public Usage  | Model with Private Usage | Model without Usage |
|----------------------------|--------------------------|--------------------------|---------------------|
| `@access(public)`          | Public                   | Public                   | Public              |
| `@access(private)(default)`| Public                   | Private                  | Private             |
```

### Checklist

- [X] Follow our [Code of Conduct](https://github.com/azure/typespec-azure/blob/main/CODE_OF_CONDUCT.md)
- [X] Check that this issue is about the Azure libraries for typespec. For feature request in the typespec language or core libraries file it in the [TypeSpec repo](https://github.com/Microsoft/TypeSpec/issues/new/choose)
- [X] Read the [docs](https://azure.github.io/typespec-azure/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by joheredi**
Fixes #1299

### CODE REVIEW

**Review by iscai-msft — APPROVED** [review]

File: packages/typespec-client-generator-core/lib/decorators.tsp [hunk 1]
```diff
@@ -260,6 +260,9 @@ enum Access {
 
 /**
  * Set explicit access for operations, models and enums.
+ * When setting access for namespaces,
+ * the access the access info will be propagated to the model children of the namespace
+ * if the model has an access override, the model override takes precedence.
```
Comment (tadelesh) [inline packages/typespec-client-generator-core/lib/decorators.tsp:265]:
```suggestion
 * the access info will be propagated to the models defined in the namespace.
 * If the model has an access override, the model override takes precedence.
```

File: packages/typespec-client-generator-core/src/decorators.ts [hunk 2]
```diff
@@ -951,17 +951,23 @@ export const $access: AccessDecorator = (
...
+
+  return accessOverride;
 }
 
 export function getAccess(
   context: TCGCContext,
-  entity: Model | Enum | Operation | Union
-): AccessFlags {
+  entity: Model | Enum | Operation | Union | Namespace
+) {
```
Comment (tadelesh) [inline packages/typespec-client-generator-core/src/decorators.ts:968]: this func is used for get final calculation result of access, so i don't think we should add namespace

File: packages/typespec-client-generator-core/src/decorators.ts [hunk 3]
```diff
@@ -951,17 +951,23 @@ export const $access: AccessDecorator = (
...
   const override = getScopedDecoratorData(context, accessKey, entity);
```
Comment (tadelesh) [inline packages/typespec-client-generator-core/src/decorators.ts:969]:
```suggestion
  const override = getAccessOverride(context, entity);
```

File: packages/typespec-client-generator-core/src/decorators.ts [hunk 4]
```diff
@@ -951,17 +951,23 @@ export const $access: AccessDecorator = (
...
-  if (override || entity.kind === "Operation") {
+  if (override) {
     return override || "public";
```
Comment (tadelesh) [inline packages/typespec-client-generator-core/src/decorators.ts:971]:
```suggestion
    return override;
```

File: packages/typespec-client-generator-core/src/decorators.ts [hunk 5]
```diff
@@ -970,15 +976,20 @@ export function getAccess(
...
-    case "Union":
+    case "Union": {
       const type = getSdkUnion(context, entity);
       if (type.kind === "enum" || type.kind === "model") {
         return type.access;
       }
       return "public";
+    }
+    case "Operation":
+    case "Namespace":
```
Comment (tadelesh) [inline packages/typespec-client-generator-core/src/decorators.ts:987]:
```suggestion
    case "Operation":
```

File: packages/typespec-client-generator-core/test/decorators.test.ts [hunk 6]
```diff
@@ -1690,6 +1690,36 @@ describe("typespec-client-generator-core: decorators", () => {
...
+          @test
+          model Test {
+            prop: string;
+          }
+        `)) as { Test: Operation };
+  
+        const actual = getAccess(runner.context, Test);
+        strictEqual(actual, "internal");
+      })
+    })
```
Comment (tadelesh) [inline packages/typespec-client-generator-core/test/decorators.test.ts:1722]: also need test for operation

**Review by tadelesh — CHANGES_REQUESTED** [review]

**Review by tadelesh — APPROVED** [review]

File: packages/typespec-client-generator-core/test/decorators.test.ts (hunk 6 above)
Comment (joheredi) [reply packages/typespec-client-generator-core/test/decorators.test.ts]: Added additional tests

**Pull request merged**